admin_queries/course_instance_compute_time.sql

```diff
@@ -1,83 +1,97 @@
-        JOIN course_instances AS ci ON (ci.id = a.course_instance_id)
-        JOIN pl_courses AS c ON (c.id = ci.course_id)
-        JOIN institutions AS i ON (i.id = c.institution_id)
+      workspaces AS w
+      JOIN variants AS v ON (v.workspace_id = w.id)
+      JOIN instance_questions AS iq ON (iq.id = v.instance_question_id)
+      JOIN assessment_instances AS ai ON (ai.id = iq.assessment_instance_id)
+      JOIN assessments AS a ON (a.id = ai.assessment_id)
+      JOIN course_instances AS ci ON (ci.id = a.course_instance_id)
+      JOIN pl_courses AS c ON (c.id = ci.course_id)
+      JOIN institutions AS i ON (i.id = c.institution_id)
     WHERE
-        ($institution_short_name = '' OR i.short_name = $institution_short_name)
-        AND w.created_at BETWEEN $start_date AND $end_date
+      (
+        $institution_short_name = ''
+        OR i.short_name = $institution_short_name
+      )
+      AND w.created_at BETWEEN $start_date AND $end_date
     GROUP BY
-        ci.id
-),
-grading_job_durations AS (
+      ci.id
+  ),
+  grading_job_durations AS (
     SELECT
-        ci.id AS course_instance_id,
-        COUNT(DISTINCT gj.id) AS grading_job_count,
-        sum(extract(epoch from gj.grading_finished_at - gj.grading_received_at) / 3600.0) AS duration_hours
+      ci.id AS course_instance_id,
+      COUNT(DISTINCT gj.id) AS grading_job_count,
+      sum(
+        extract(
+          epoch
+          from
+            gj.grading_finished_at - gj.grading_received_at
+        ) / 3600.0
+      ) AS duration_hours
     FROM
-        grading_jobs AS gj
-        JOIN submissions AS s ON (s.id = gj.submission_id)
-        JOIN variants AS v ON (v.id = s.variant_id)
-        JOIN instance_questions AS iq ON (iq.id = v.instance_question_id)
-        JOIN assessment_instances AS ai ON (ai.id = iq.assessment_instance_id)
-        JOIN assessments AS a ON (a.id = ai.assessment_id)
-        JOIN course_instances AS ci ON (ci.id = a.course_instance_id)
-        JOIN pl_courses AS c ON (c.id = ci.course_id)
-        JOIN institutions AS i ON (i.id = c.institution_id)
+      grading_jobs AS gj
+      JOIN submissions AS s ON (s.id = gj.submission_id)
+      JOIN variants AS v ON (v.id = s.variant_id)
+      JOIN instance_questions AS iq ON (iq.id = v.instance_question_id)
+      JOIN assessment_instances AS ai ON (ai.id = iq.assessment_instance_id)
+      JOIN assessments AS a ON (a.id = ai.assessment_id)
+      JOIN course_instances AS ci ON (ci.id = a.course_instance_id)
+      JOIN pl_courses AS c ON (c.id = ci.course_id)
+      JOIN institutions AS i ON (i.id = c.institution_id)
     WHERE
-        ($institution_short_name = '' OR i.short_name = $institution_short_name)
-        AND gj.date BETWEEN $start_date AND $end_date
-        AND gj.grading_method = 'External'
+      (
+        $institution_short_name = ''
+        OR i.short_name = $institution_short_name
+      )
+      AND gj.date >= $start_date
+      AND gj.date <= $end_date
```

Comment:
For some reason I had to replace this BETWEEN with comparisons. I believe it has do to with a corner case in sql-formatter since `date` is a keyword, since other cases in this same query that use BETWEEN are still ok.

Reply:
Ok, I think I found the reason: https://github.com/sql-formatter-org/sql-formatter/issues/534

Reply:
TLDR to anyone reading this and not the Slack discussion: any constructs like `a BETWEEN b AND c AND d` need to be rewritten as `(a BETWEEN b AND c) AND d` to avoid ambiguity. It had nothing to do with the column name.